git-extra/git-sdk.sh

```diff
@@ -65,13 +65,22 @@ sdk () {
 		'') break;;
 		-*) sdk die "Unknown option: %s\n" "$1"; return 1;;
 		esac; do shift; done &&
+		{ cat >"$TMP/get-desktop-location.vbs" <<-\EOF
+		Dim objShell
+		Dim strPath
 
+		Set objShell = Wscript.CreateObject("Wscript.Shell")
+		strPath = objShell.SpecialFolders("Desktop")
+		wscript.echo strPath
+		EOF
+		} &&
+		path_to_desktop=$(cscript //nologo $TMP/get-desktop-location.vbs) &&
 		case "$(uname -m)" in
 		i686) bitness=" 32-bit";;
 		x86_64) bitness=" 64-bit";;
 		*) bitness=;;
 		esac &&
-		desktop_icon_path="$USERPROFILE/Desktop/Git SDK$bitness.lnk" &&
+		desktop_icon_path="$path_to_desktop/Git SDK$bitness.lnk" &&
 		if test -n "$force" || test ! -f "$desktop_icon_path"
 		then
 			create-shortcut.exe --icon-file /msys2.ico --work-dir \
```

Comment:
We already use the `create-shortcut.exe` that is compiled in the very same package... Would it not make more sense to add an option to that executable that tells it to interpret the path as relative to the Desktop, and then use [`SHGetKnownFolderPath()`](https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath) directly in C?

That would avoid requiring Visual Basic Script, and make `create-shortcut.exe` more useful.

Reply:
That would make the `create-shortcut` more specialized. 

At the moment it just creates a shortcut to a location, with a workdir and an icon.

There's no  way in the Windows API do something like this:

Expand `shell:desktop\git-bash.lnk` do `C:\Users\KM\OneDrive\Desktop\git-bash.lnk`

We can fix it by offering a full-path option (which is what's there now) and amend so that it would take a `KNOWNFOLDERID` and a relative location.

Reply:
What I meant was `create-shortcut.exe --desktop-shortcut [...] "Git SDK$bitness.lnk"`.

Reply:
... or more precisely, what I meant was this:

```diff
diff --git a/git-extra/create-shortcut.c b/git-extra/create-shortcut.c
index 81ebce9fd..c0472e2f1 100644
--- a/git-extra/create-shortcut.c
+++ b/git-extra/create-shortcut.c
@@ -25,7 +25,7 @@ int main(int argc, char **argv)
 	const char *progname = argv[0];
 	const char *work_dir = NULL, *arguments = NULL, *icon_file = NULL;
 	const char *description = NULL;
-	int show_cmd = 1;
+	int show_cmd = 1, desktop_shortcut = 0;
 
 	static WCHAR wsz[1024];
 	HRESULT hres;
@@ -45,7 +45,12 @@ int main(int argc, char **argv)
 			icon_file = argv[2];
 		else if (!strcmp(argv[1], "--description"))
 			description = argv[2];
-		else {
+		else if (!strcmp(argv[1], "--desktop-shortcut")) {
+			desktop_shortcut = 1;
+			argc--;
+			argv++;
+			continue;
+		} else {
 			fprintf(stderr, "Unknown option: %s\n", argv[1]);
 			return 1;
 		}
@@ -92,9 +97,30 @@ int main(int argc, char **argv)
 	if (description)
 		psl->lpVtbl->SetDescription(psl, description);
 
-	wsz[0] = 0;
-	MultiByteToWideChar(CP_ACP,
-			0, argv[2], -1, wsz, 1024);
+	if (!desktop_shortcut)
+		wsz[0] = 0;
+	else {
+		PWSTR p;
+
+		hres = SHGetKnownFolderPath(&FOLDERID_Desktop,
+					    KF_FLAG_DONT_UNEXPAND, NULL, &p);
+		check_hres(hres, "Could not get desktop path");
+
+		desktop_shortcut = wcslen(p);
+		if (desktop_shortcut + 2 + strlen(argv[2]) * 3 >
+		    sizeof(wsz) / sizeof(WCHAR)) {
+			fwprintf(stderr,
+				 L"Error: Too long Desktop path: %s\n", p);
+			exit(1);
+		}
+		memcpy(wsz, p, sizeof(WCHAR) * desktop_shortcut);
+		wsz[desktop_shortcut++] = L'\\';
+		wsz[desktop_shortcut] = L'\0';
+		CoTaskMemFree(p);
+	}
+	MultiByteToWideChar(CP_ACP, 0, argv[2], -1,
+			    wsz + desktop_shortcut,
+			    sizeof(wsz) / sizeof(WCHAR) - desktop_shortcut);
 	hres = ppf->lpVtbl->Save(ppf,
 			(const WCHAR*)wsz, TRUE);
 
```

Reply:
I now pushed this to the [`improve-desktop-location-detection`](https://github.com/dscho/build-extra/commits/improve-desktop-location-detection) branch in my fork.

Reply:
Welp, my C/C++ is not at the level that I'd feel comfortable making a contribution like that.

Reply:
You don't have to, you can just cherry-pick the two top commits of the branch I pushed. Then add your commit on top, using `--desktop-shortcut` instead of the Visual Basic scriptlet.